### PR TITLE
Add Resource Reference Documentation

### PR DESCRIPTION
**Pull request by djosborne**
For #81

### CODE REVIEW

File: docs/calicoctl/resources/policy.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    protocol: tcp
+    icmp:
+      - type: 10
+      - code: 6
+    "!protocol": ipv6
+    "!icmp":
+      - type: 19
+      - code: 255
+    source:
+      tag: footag
```
Comment (djosborne) [inline docs/calicoctl/resources/policy.md]: @robbrockbank Am I right here? [Tags are just strings](https://github.com/tigera/libcalico-go/blob/master/lib/api/rule.go#L46). Is this a legitimate sample?

File: docs/calicoctl/resources/workloadendpoint.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+| name           | The name of this endpoint resource.                        | Required for `create`/`update`/`delete`. | string |
+| workloadID     | The unique ID of the workload this endpoint belongs to.    | Required for `create`/`update`/`delete`. | string |
+| orchestratorID | The ID of the orchestrator that created this endpoint.     | Required for `create`/`update`/`delete`. | string |
+| hostname       | The hostname of the host where this endpoint resides.      | Required for `create`/`update`/`delete`. | string |
+| labels         | A set of labels to apply to this endpoint.                 |      | Dictionary with key and values as strings. |
+
+#### Spec
+| name          | description                                             | requirements                | schema          |
+|---------------|---------------------------------------------------------|-----------------------------|-----------------|
+| ipNetworks    | TODO. | | List of strings |
```
Comment (djosborne) [inline docs/calicoctl/resources/workloadendpoint.md]: @robbrockbank Can you help with a descrip on this one?

File: docs/calicoctl/resources/policy.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+| ingress  | The ingress rules belonging to this policy.                          | | List of [RuleSpecs](#rulespec) |
+| egress   | The egress rules belonging to this policy.                           | | List of [RuleSpecs](#rulespec)  |
+| selector | Selector expression.                                                 | See [selector expression documentation](http://docs.projectcalico.org/en/latest/etcd-data-model.html#tiered-security-policy) | string |
+
+#### RuleSpec
+| name        | description                                | requirements | schema |
+|-------------|--------------------------------------------|----------------|--------|
+| action      | Action to perform when matching this rule.  Can be one of: `nextTier`, `allow`, `deny` |  | string |
+| protocol    | Positive protocol match.  | Can be one of: `tcp`, `udp`, `icmp`, `icmpv6`, `sctp`, `udplite`, or an integer 1-255. | string |
+| icmpType    | Positive match on ICMP type. | Can be integer 1-255 | integer |
```
Comment (gunjan5) [inline docs/calicoctl/resources/policy.md]: this should be `icmp` and subtype `type` and `code` (`icmpType` , `icmpCode`, `!icmpType` and `!icmpCode` are no longer the keywords as of #93 ) 

```yaml
icmp:
    - type: 10
    - code: 6
```

File: docs/calicoctl/resources/profile.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+| tags    | A list of tag names to apply to endpoints using this profile. Note that tags are expected to be deprecated shortly in preference to labels. Label with blank value is equivalent to tags. | | list of strings |
+| ingress  | The ingress rules belonging to this policy.                          | | List of [RuleSpecs](#rulespec) |
+| egress   | The egress rules belonging to this policy.                           | | List of [RuleSpecs](#rulespec)  |
+
+#### RuleSpec
+| name        | description                                | requirements | schema |
+|-------------|--------------------------------------------|----------------|--------|
+| action      | Action to perform when matching this rule.  Can be one of: `nextTier`, `allow`, `deny` |  | string |
+| protocol    | Positive protocol match.  | Can be one of: `tcp`, `udp`, `icmp`, `icmpv6`, `sctp`, `udplite`, or an integer 1-255. | string |
+| icmpType    | Positive match on ICMP type. | Can be integer 1-255 | integer |
```
Comment (gunjan5) [inline docs/calicoctl/resources/profile.md]: Same as the other comment :)

File: docs/calicoctl/resources/policy.md (hunk 1 above)
Comment (robbrockbank) [inline docs/calicoctl/resources/policy.md]: Yes.   Profiles have a list of tags associated with them, and endpoints assigned to use a profile will have the profiles tags associated with the endpoint.

File: docs/calicoctl/resources/workloadendpoint.md (hunk 2 above)
Comment (robbrockbank) [inline docs/calicoctl/resources/workloadendpoint.md]: These are the CIDRs assigned to the interface.

File: docs/calicoctl/resources/policy.md (hunk 1 above)
Comment (djosborne) [inline docs/calicoctl/resources/policy.md]: So is it a `list of strings` or a `string`?

Comment (gunjan5) [inline docs/calicoctl/resources/policy.md]: I think it's a `string` according to the struct definition.
https://github.com/tigera/libcalico-go/blob/master/lib/api/rule.go#L46

**Pull request merged**